### DISCUSSION TITLE
Johansen Cointegration Test

### DISCUSSION THREAD

**Discussion by cdrnet** [mathnet/mathnet-numerics#794] [Ideas]
From: https://mathnetnumerics.codeplex.com/workitem/5730

> Would like to see support for Johansen test. 
> 
> http://www.mathworks.com/help/econ/jcitest.html 
> http://en.wikipedia.org/wiki/Johansen_test

**Comment by jaredbroad** [comment]
Bump. Any plans @cdrnet? @mchandschuh

**Comment by cdrnet** [comment]
How would an API design for this ideally look like, that would support your use case in a clean way?

**Comment by mchandschuh** [comment]
Ideally the API would support usage similar to this [python](https://github.com/josef-pkt/statsmodels/blob/coint/statsmodels/tsa/johansen.py) implementation.

Following along the lines of the `Correlation` class, I could see something like the following in C#:

``` C#
var result = Cointegration.Johansen(Matrix<double> matrix, int order, int laggedDifferences);
```

**Comment by glyphard** [comment]
This suggested API makes sense.

FWIW, the suggested api is reasonably consistent with the R package for
Johansen Cointegration Test (urca).  It is a little bit different than the
matlab econometrics toolbox version of Johansen Cointegration, but that is
probably ok because the matlab api is older and offers a lot more complex
control over the implementation details which likely wouldn't be built into
a first version of JCI in mathnet.

**Comment by Levitikon217** [comment]
Any updates on this? It would be really cool if any Cointegration support was added.  Johansen or Augmented Dickey Fuller (ADF)!

**Comment by karuzzo** [comment]
@cdrnet Hi, I have my custom implementations of the ADF test and the Johansen test using your library, I don’t know how correct these implementations are, but I’m ready to send them to you so that you check them and correct them if necessary so that these tests can be added to your library.

**Reply by usanovserg** [reply]
Hi! Karuzzo, how can I contact you?
